api/src/main/java/io/opentelemetry/metrics/SyncInstrument.java

```diff
@@ -26,7 +26,7 @@
  * @since 0.3.0
  */
 @ThreadSafe
-public interface InstrumentWithBinding<B extends BoundInstrument> extends Instrument {
+public interface SyncInstrument<B extends BoundInstrument> extends Instrument {
```

Comment:
I think I'd prefer `SynchronousInstrument` here, even though it's a little longer. Also, I think some javadoc explaining what "synchronous" means in this context would be very useful.

Reply:
Done in #1202.